Week3/homework/step2-6.js

```diff
@@ -4,13 +4,21 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 function printArray2d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  for (let i = 0; i < arr.length; i++) {
+    for (let j = 0; j < arr[i].length; j++) {
+      console.log(arr[i][j]);
+    }
+  }
 }
 
 function printArray3d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  for (let i = 0; i < arr.length; i++) {
+    if (arr[i] instanceof Array) {
+      printArray3d(arr[i]);
+    } else {
+      console.log(arr[i]);
+    }
+  }
 }
```

Comment:
Excellent use of a recursive function (which we didn't cover in class). Since this recursive function can deal with arrays of any dimension the name `printArray3d` is too limiting. Can you rewrite step 2-6 such that you still have two functions `printArray2d` and `printArray3d` (as required to pass the unit tests) but that both make use of your recursive function and thus make your code DRY?